conformance/lang/expressions/literals/string_literal.balt

```diff
@@ -0,0 +1,330 @@
+Test-Case: output
+Description: Test basic string literals.
+Labels: string
+
+function init() {
+    string a = "Ballerina";
+    io:println(a); // @output Ballerina
+
+    string b = "1234";
+    io:println(b); // @output 1234
+
+    string c = "{name: Kevin}";
+    io:println(c); // @output {name: Kevin}
+
+    io:println("+_)(*&^%$#@!"); // @output +_)(*&^%$#@!
+}
+
+Test-Case: parser-error
+Description: Test invalid usage of line break in string literal.
+Labels: string
+
+function init() {
+    string _ = "First String
+                Second String"; // @error line break is not allowed in string value
+}
+
+Test-Case: parser-error
+Description: Test invalid usage of `\` in string literal.
+Labels: string
+
+function init() {
+    string _ = "\"; // @error invalid usage of backslash
+
+    string _ = "Some text\"; // @error invalid usage of backslash
+}
+
+Test-Case: error
+Description: Test invalid string escape sequence.
+Labels: string
+
+function init() {
+    string _ = "Hello \Ballerina"; // @error `\B` is not a valid escape sequence
+}
+
+Test-Case: parser-error
+Description: Test invalid usage of `"` in string literal.
+Labels: string
+
+function init() {
+    string _ = """; // @error double quote is not allowed in string value
+
+    string _ = "First text " Second text"; // @error double quote is not allowed in string value
+
+    string _ = "This is double quote " "; // @error double quote is not allowed in string value
+}
+
+Test-Case: output
+Description: Test string single escape.
+Labels: string, StringSingleEscape
+
+function init() {
+    io:println("Hello \t world"); // @output Hello 	 world
+    io:println("Hello\tworld"); // @output Hello	world
+    io:println("Hello\t world"); // @output Hello	 world
+
+    io:println("Hello \nworld!"); // @output Hello 
+                                  // @output world!
+    io:println("Hello\nworld!"); // @output Hello
+                                  // @output world!
+    io:println("Hello\n world!"); // @output Hello
+                                  // @output  world!
+
+    io:println("12345 world \rHello"); // @output Hello world 
+    io:println("12345 world\rHello"); // @output Hello world
+    io:println("12345 world\r Hello"); // @output  Hello world
+
```

Comment:
Output is incorrect.

Reply:
Resolving as per the offline discussion.